core/src/clients/snyk/mod.rs

```diff
@@ -0,0 +1,22 @@
+use thiserror::Error;
+
+mod snyk_data_model;
+pub use snyk_data_model::*;
+
+mod snyk_client;
+pub use snyk_client::*;
+
+#[derive(Error, Debug)]
+pub enum SnykProviderError {
+    #[error("error connecting to Snyk: {0}")]
+    SnykConnection(String),
+
+    #[error("Invalid data recieved from Snyk: {0}")]
+    SnykDataValidationError(String),
```

Comment:
I like how you omitted Error from the First enum value here.  Keeps it kind of DRY.